src/test/java/com/microsoft/sqlserver/jdbc/datatypes/BulkCopyWithSqlVariantTest.java

```diff
@@ -0,0 +1,549 @@
+        }
+
+    }
+
+    /**
+     * Read GUID stored in SqlVariant
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void bulkCopyTestReadGUID() throws SQLException {
+        String col1Value = "1AE740A2-2272-4B0F-8086-3DDAC595BC11";
+        beforeEachSetup("uniqueidentifier", "'" + col1Value + "'");
+        SQLServerResultSet rs = (SQLServerResultSet) stmt.executeQuery("SELECT * FROM " + tableName);
+
+        SQLServerBulkCopy bulkCopy = new SQLServerBulkCopy(con);
+        bulkCopy.setDestinationTableName(destTableName);
+        bulkCopy.writeToServer(rs);
+
+        rs = (SQLServerResultSet) stmt.executeQuery("SELECT * FROM " + destTableName);
+        while (rs.next()) {
+            assertEquals("" + rs.getUniqueIdentifier(1), col1Value);
+
+        }
+    }
+
+    /**
+     * Read VarChar8000 from SqlVariant
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void bulkCopyTestVarchar8000() throws SQLException {
+        StringBuffer buffer = new StringBuffer();
+        for (int i = 0; i < 8000; i++) {
+            buffer.append("a");
+        }
+        String col1Value = buffer.toString();
+        beforeEachSetup("varchar(8000)", "'" + col1Value + "'");
+
+        SQLServerResultSet rs = (SQLServerResultSet) stmt.executeQuery("SELECT * FROM " + tableName);
+
+        SQLServerBulkCopy bulkCopy = new SQLServerBulkCopy(con);
+        bulkCopy.setDestinationTableName(destTableName);
+        bulkCopy.writeToServer(rs);
+
+        rs = (SQLServerResultSet) stmt.executeQuery("SELECT * FROM " + destTableName);
+        while (rs.next()) {
+            assertEquals(rs.getString(1), col1Value);
+        }
+    }
+
+    private void beforeEachSetup(String colType,
+            Object colValue) throws SQLException {
+        Utils.dropTableIfExists(tableName, stmt);
+        Utils.dropTableIfExists(destTableName, stmt);
+        stmt.executeUpdate("create table " + tableName + " (col1 sql_variant)");
+        stmt.executeUpdate("INSERT into " + tableName + "(col1) values (CAST (" + colValue + " AS " + colType + ") )");
+        stmt.executeUpdate("create table " + destTableName + " (col1 sql_variant)");
+    }
+
+    /**
+     * Prepare test
+     * 
+     * @throws SQLException
+     * @throws SecurityException
+     * @throws IOException
+     */
+    @BeforeAll
+    public static void setupHere() throws SQLException, SecurityException, IOException {
+        con = (SQLServerConnection) DriverManager.getConnection(connectionString);
+        stmt = con.createStatement();
+    }
+
+    /**
+     * drop the tables
+     * 
+     * @throws SQLException
+     */
+    @AfterAll
+    public static void afterAll() throws SQLException {
+        Utils.dropTableIfExists(tableName, stmt);
+        Utils.dropTableIfExists(destTableName, stmt);
+
+        if (null != stmt) {
+            stmt.close();
+        }
+        if (null != con) {
+            con.close();
+        }
+    }
+}
```

Comment:
let's also close all the closeable objects, e.g. SQLServerResultSet SQLServerBulkCopy